cmd/ingester/app/flags.go

```diff
@@ -54,13 +56,16 @@ const (
 	DefaultParallelism = 1000
 	// DefaultEncoding is the default span encoding
 	DefaultEncoding = EncodingProto
+	// DefaultMaxReadsPerSecond is the default max reads per second for the span processor
+	DefaultMaxReadsPerSecond = 0.0
```

Comment:
semantically this is wrong; if you pass in 0, you'd expect the consumer to do no work but this value does the exact opposite. Create a new const `DisableRateLimit = 0.0` in rate_limiter.go and use that here instead.

Reply:
you can also consider value 0 to be "undefined" and thus not causing any rate limit. Certainly from the CLI API point of view, you should not need to pass any flags to run without a rate limit, and if you do want a rate limit you pass a non-0 value.

Reply:
I was using `-1` for this, but used `0` instead seeing as Go default initialization makes this hard to get wrong. I do not mind using another value if you'd like. Not sure what makes the most sense here.